langwatch/src/components/datasets/DatasetMapping.tsx

```diff
@@ -507,29 +508,11 @@ export const TracesMapping = ({
     }
 
     for (const trace of expandedTraces) {
-      const entry = Object.fromEntries(
```

Comment:
how come this piece was not moved along?

```
    for (const expansion of expansions) {
      expandedTraces = expandedTraces.flatMap((trace) =>
        TRACE_EXPANSIONS[expansion].expansion(trace)
      );
```

you are just passing `expansions` as a set but never really expanding it as the above for loop when the trigger is triggered

please move this and the whole for-loop along, and do a proper testing of an add do dataset trigger with expansions